File: examples/performance/prime/prime.c
#include <stdlib.h>
#include <stdio.h>
#include <math.h>

#define TRUE 1
#define FALSE 0

int isPrime(int num)
{
   if (num == 2)
   {
      return TRUE;
   }

   int found = TRUE;
   for (int i = 2; i <= sqrt(num) + 1; ++i)
   {
      if (num % i == 0)
      {
         found = FALSE;
         break;
      }
   }

   return found;
}

int main(int argc, char **argv)
{
   int MAX_I = 100;
   int primes = 0;

   if (argc > 1)
   {
      MAX_I = atoi(argv[1]);
   }

   for (int i = 2; i < MAX_I; ++i)
   {
      if (isPrime(i))
      {
         ++primes;
      }
   }

   printf("%d\n", primes);
}

File: examples/performance/prime/prime.py
import math
import sys

def isPrime(num):
    found = True
    for i in range(2, int(math.sqrt(num) + 1)):
        if (num % i == 0):
            found = False
            break

    return found

MAX_I = 10000
if len(sys.argv) > 1:
    MAX_I = int(sys.argv[1])
    primes = 0

for i in range(2, MAX_I):
    if isPrime(i):
        primes += 1

print(primes)
